### ISSUE TITLE
possibly uninitialized variable in Pruner

### ISSUE THREAD

**Issue by malb** [fplll/fplll#278]
```
  CXX      pruner.lo
In file included from defs.h:56:0,
                 from bkz_param.h:23,
                 from pruner.h:19,
                 from pruner.cpp:16:
nr/dpe.h: In member function ‘int fplll::Pruner<FT>::nelder_mead(fplll::Pruner<FT>::evec&) [with FT = fplll::FP_NR<dpe_struct [1]>; fplll::Pruner<FT>::evec = std::array<fplll::FP_NR<dpe_struct [1]>, 1023ul>]’:
nr/dpe.h:365:28: warning: ‘fs_maxi_last.dpe_struct::exp’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   DPE_EXP(x)  = DPE_EXP(y) + DPE_EXP(z);
                            ^
pruner.cpp:620:6: note: ‘fs_maxi_last.dpe_struct::exp’ was declared here
   FT fs_maxi_last;  // value of the last centroid
      ^
In file included from defs.h:56:0,
                 from bkz_param.h:23,
                 from pruner.h:19,
                 from pruner.cpp:16:
nr/dpe.h:364:29: warning: ‘fs_maxi_last.dpe_struct::d’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   DPE_MANT(x) = DPE_MANT(y) * DPE_MANT(z);
                             ^
pruner.cpp:620:6: note: ‘fs_maxi_last.dpe_struct::d’ was declared here
   FT fs_maxi_last;  // value of the last centroid
      ^
In file included from ../fplll/nr/nr.h:19:0,
                 from ../fplll/nr/numvect.h:19,
                 from nr/matrix.h:21,
                 from gso_interface.h:21,
                 from gso.h:21,
                 from pruner.h:21,
                 from pruner.cpp:16:
../fplll/nr/nr_FP_ld.inl: In member function ‘int fplll::Pruner<FT>::nelder_mead(fplll::Pruner<FT>::evec&) [with FT = fplll::FP_NR<long double>; fplll::Pruner<FT>::evec = std::array<fplll::FP_NR<long double>, 1023ul>]’:
../fplll/nr/nr_FP_ld.inl:187:17: warning: ‘fs_maxi_last.fplll::FP_NR<long double>::data’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   data = b.data * c.data;
                 ^
pruner.cpp:620:6: note: ‘fs_maxi_last.fplll::FP_NR<long double>::data’ was declared here
   FT fs_maxi_last;  // value of the last centroid
      ^
In file included from ../fplll/nr/nr.h:18:0,
                 from ../fplll/nr/numvect.h:19,
                 from nr/matrix.h:21,
                 from gso_interface.h:21,
                 from gso.h:21,
                 from pruner.h:21,
                 from pruner.cpp:16:
../fplll/nr/nr_FP_d.inl: In member function ‘int fplll::Pruner<FT>::nelder_mead(fplll::Pruner<FT>::evec&) [with FT = fplll::FP_NR<double>; fplll::Pruner<FT>::evec = std::array<fplll::FP_NR<double>, 1023ul>]’:
../fplll/nr/nr_FP_d.inl:184:17: warning: ‘fs_maxi_last.fplll::FP_NR<double>::data’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   data = b.data * c.data;
                 ^
pruner.cpp:620:6: note: ‘fs_maxi_last.fplll::FP_NR<double>::data’ was declared here
   FT fs_maxi_last;  // value of the last centroid
      ^
```

[source](https://travis-ci.org/fplll/fplll/jobs/253131702)

**Comment by lducas** [comment]
Hum, compilation on my side give me now such warning. I intialized the variable in the improve_pruner branch though.

**Comment by malb** [comment]
Looks like this was fixed in 17cb7f903b0cf87987a87b703bfc3c6edf15df78
